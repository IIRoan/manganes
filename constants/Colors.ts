// constants/Colors.ts

export const Colors = {
  light: {
    primary: '#2E8B57',
    background: '#F5F5F5',
    card: '#FFFFFF',
    text: '#333333',
    border: '#E0E0E0',
    notification: '#FF6B6B',
    tint: '#2E8B57',
    tabIconDefault: '#BDBDBD',
    tabIconSelected: '#2E8B57',
    secondaryText: '#757575',
    error: '#FF6B6B',

  },
  dark: {
    primary: '#4CAF50',
    background: '#121212',
    card: '#1f1f1d',
    text: '#E0E0E0',
    border: '#333333',
    notification: '#FF6B6B',
    tint: '#4CAF50',
    tabIconDefault: '#757575',
    tabIconSelected: '#4CAF50',
    secondaryText: '#BDBDBD',
    error: '#FF6B6B',
  },
};

export type ColorScheme = keyof typeof Colors;
export type ThemeColors = typeof Colors.light & typeof Colors.dark;
