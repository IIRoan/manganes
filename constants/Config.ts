export const MANGA_API_URL = 'https://mangafire.to';
