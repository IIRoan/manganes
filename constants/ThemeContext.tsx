import React, { createContext, useState, useContext, useEffect } from 'react';
import { useColorScheme } from 'react-native';
import { ColorScheme } from '@/constants/Colors';

type Theme = 'light' | 'dark' | 'system';

interface ThemeContextType {
  theme: Theme;
  systemTheme: ColorScheme;
  setTheme: (theme: Theme) => void;
  toggleTheme: () => void;
  actualTheme: 'light' | 'dark';
}

const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

export const ThemeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [theme, setTheme] = useState<Theme>('system');
  const systemColorScheme = useColorScheme() as ColorScheme;

  const toggleTheme = () => {
    setTheme(prevTheme => {
      switch (prevTheme) {
        case 'light': return 'dark';
        case 'dark': return 'system';
        case 'system': return 'light';
      }
    });
  };

  const actualTheme = theme === 'system' ? systemColorScheme : theme;

  return (
    <ThemeContext.Provider value={{ 
      theme, 
      systemTheme: systemColorScheme, 
      setTheme,
      toggleTheme,
      actualTheme
    }}>
      {children}
    </ThemeContext.Provider>
  );
};

export const useTheme = () => {
  const context = useContext(ThemeContext);
  if (context === undefined) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
};