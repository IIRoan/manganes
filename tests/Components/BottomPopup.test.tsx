import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import BottomPopup from '@/components/BottomPopup';
import { useTheme } from '@/constants/ThemeContext';

jest.mock('@/constants/ThemeContext', () => ({
  useTheme: jest.fn(),
}));

jest.mock('@expo/vector-icons', () => ({
  Ionicons: 'Icon',
}));

jest.mock('react-native-reanimated', () => {
  // Mock the Reanimated module to avoid errors during testing
  const Reanimated = require('react-native-reanimated/mock');
  Reanimated.default.call = () => {};
  return Reanimated;
});

jest.mock('react-native-safe-area-context', () => ({
  useSafeAreaInsets: () => ({ bottom: 0 }),
}));

describe('BottomPopup Component', () => {
  const onCloseMock = jest.fn();
  const optionPressMock = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    (useTheme as jest.Mock).mockReturnValue({
      theme: 'light',
      systemTheme: 'light',
    });
  });

  it('calls onClose when close button is pressed', () => {
    const { getByTestId } = render(
      <BottomPopup
        visible={true}
        title="Test Popup"
        onClose={onCloseMock}
        options={[]}
      />
    );

    fireEvent.press(getByTestId('close-button'));
    expect(onCloseMock).toHaveBeenCalled();
  });

  it('calls option onPress and onClose when an option is pressed', () => {
    const { getByText } = render(
      <BottomPopup
        visible={true}
        title="Test Popup"
        onClose={onCloseMock}
        options={[
          { text: 'Option 1', onPress: optionPressMock },
        ]}
      />
    );

    fireEvent.press(getByText('Option 1'));
    expect(optionPressMock).toHaveBeenCalled();
    expect(onCloseMock).toHaveBeenCalled();
  });

  it('does not render when not visible', () => {
    const { queryByText } = render(
      <BottomPopup
        visible={false}
        title="Hidden Popup"
        onClose={onCloseMock}
        options={[]}
      />
    );

    expect(queryByText('Hidden Popup')).toBeNull();
  });

  it('renders options with icons when provided', () => {
    const { getByText } = render(
      <BottomPopup
        visible={true}
        title="Icon Options"
        onClose={onCloseMock}
        options={[
          { text: 'Option with Icon', onPress: optionPressMock, icon: 'options' },
        ]}
      />
    );

    expect(getByText('Option with Icon')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const tree = render(
      <BottomPopup
        visible={true}
        title="Snapshot Popup"
        onClose={onCloseMock}
        options={[
          { text: 'Snapshot Option', onPress: optionPressMock },
        ]}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
