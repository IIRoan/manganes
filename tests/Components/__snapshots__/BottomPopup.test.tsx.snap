// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomPopup Component matches snapshot 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "rgba(0,0,0,0.5)",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "opacity": 0,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
            "elevation": 5,
            "maxHeight": "80%",
            "paddingBottom": 24,
            "paddingHorizontal": 24,
            "paddingTop": 12,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": -3,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 5,
          },
          {
            "transform": [
              {
                "translateY": 1334,
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          {
            "alignSelf": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 2,
            "height": 4,
            "marginBottom": 16,
            "width": 40,
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "#333333",
              "fontSize": 22,
              "fontWeight": "bold",
            }
          }
          testID="bottom-popup-title"
        >
          Snapshot Popup
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#F5F5F5",
              "borderRadius": 20,
              "opacity": 1,
              "padding": 8,
            }
          }
          testID="close-button"
        >
          <Icon
            color="#333333"
            name="close"
            size={24}
          />
        </View>
      </View>
      <View
        style={
          {
            "marginTop": 8,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderBottomColor": "#E0E0E0",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "opacity": 1,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#333333",
                "flex": 1,
                "fontSize": 18,
              }
            }
          >
            Snapshot Option
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
